packages/core/botpress/tests/bot.js

```diff
@@ -0,0 +1,127 @@
+const botConfig = path.resolve(__dirname, '../tmp-bot/config')
+const botpressDir = path.resolve(__dirname)
+const rootDir = path.resolve(__dirname, '../../../../')
+const compileLinkBash = path.resolve(rootDir, './compile-link.sh')
+const configVar = require('./config/env.json')
+
+const botpressModules = [
+  '@botpress/analytics',
+  '@botpress/audience',
+  '@botpress/broadcast',
+  '@botpress/hitl',
+  '@botpress/nlu',
+  '@botpress/qna',
+  '@botpress/scheduler',
+  '@botpress/terminal',
+  '@botpress/channel-messenger',
+  '@botpress/channel-slack',
+  '@botpress/channel-telegram',
+  '@botpress/channel-twilio',
+  '@botpress/channel-microsoft',
+  '@botpress/builtins',
+  '@botpress/util-roles',
+  '@botpress/skill-choice',
+  'botpress'
+]
+
+const LAUNCHED = /Bot launched/gi
+
+describe('Create new bot', function() {
+  this.timeout(0)
+
+  let hasError = false
+
+  const ifErrorStopDescribe = error => {
+    if (hasError || error) {
+      hasError = true
+      this.exit()
+    }
+  }
+
+  // eslint-disable-next-line
+  before(() => {
+    existsSync(botDir) && execSync(`rm -rf ${botDir}`)
+  })
+
+  it('build all modules', done => {
+    exec(`bash ${compileLinkBash}`, err => {
+      ifErrorStopDescribe(err)
+      expect(err).to.be.null
+
+      done()
+    })
+  })
+
+  it('create', done => {
+    exec(`node ${bin} init ${botDir} -y`, err => {
+      ifErrorStopDescribe(err)
+      expect(err).to.be.null
+
+      done()
+    })
+  })
+
+  it('copy config', done => {
+    exec(`rm -rf ${botConfig} && cp -a ./tests/config ${botDir}`, err => {
+      ifErrorStopDescribe(err)
+      expect(err).to.be.null
+
+      done()
+    })
+  })
+
+  it('install modules', done => {
+    exec(`yarn add ${botpressModules.join(' ')}`, { cwd: botDir }, err => {
+      ifErrorStopDescribe(err)
+      expect(err).to.be.null
+
+      done()
+    })
+  })
+
+  it('yarn link modules', done => {
+    exec(`yarn link ${botpressModules.join(' ')}`, { cwd: botDir }, err => {
+      ifErrorStopDescribe(err)
+      expect(err).to.be.null
+
+      done()
+    })
+  })
+
+  it('run bot', done => {
+    let lastDate = moment().toDate()
+    let isLaunched = false
+    const botProcess = exec('node index.js', { cwd: botDir, env: configVar })
+
+    botProcess.stdout.on('data', data => {
+      const lastLog = data.toString()
+      const hasError = /error/gi.test(lastLog)
+      const date = lastLog.match(/[0-9:]+/g)
+      isLaunched = isLaunched || LAUNCHED.test(lastLog)
+
+      lastDate = date ? moment(date[0], 'HH:mm:ss').toDate() : lastDate
+
+      expect(hasError).to.be.false
+    })
+
+    const botStateCheck = setInterval(() => {
+      const TEN_SECONDS = 10000
+      const now = moment().toDate()
+      const isEnd = now - moment(lastDate, 'HH:mm:ss').toDate() > TEN_SECONDS
+
+      if (isEnd && isLaunched) {
+        botProcess.kill('SIGINT')
+        clearInterval(botStateCheck)
+        done()
+      }
+    }, 20000)
```

Comment:
So what is the outcome of this test? We run the bot, wait until it starts and for up to 10s after the last log line and exit. But we don't seem to make any conclusion from it, or am I wrong?

Reply:
That's clear. But what is the outcome? We know that it starts and then eventually it stops printing logs. But what's the conclusion? Did we measure the efficiency? What's the result?

Reply:
@emirotin  We know that new bot will works fine. Some behaviors are pass through of modules and not always we have all needed modules installed when we make changes. That's why I created a test which can check it on start stage and prevent bot crashing. It doesn't prevent all case but it's only start. If you have suggestion please right to `dmytro@botpress.io`. Thanks)

Reply:
In this case the PR title is misleading. It doesn't test for efficiency, it's a smoke-test that checks that ht bot basically can start. Which is not bad too, it was just hard to understand.

I would though do 2 more things:
1) print the time from the start to the console before quitting the test
2) add some timeout to the test so that it eventually fails if the bot is taking more than X seconds to start

Reply:
@emirotin how you looks to 20 min timeout? because usally this test get on my PC ~14 min